src/model.rs

```diff
@@ -2,11 +2,16 @@ use std::fmt::Display;
 
 use merge::Merge;
 
-use serde::{Deserialize, Serialize};
+use serde::{
+    de::{Error as DeError, Visitor},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
 use serde_json::Value as JsonValue;
 
 use strum_macros::Display;
 
+const JSON_MAX_SAFE_INTERGER: i128 = 9007199254740991;
```

Comment:
I think it would be more suitable to use: `i64::MIN` & `i64::MAX` as the cutoff values

Reply:
Wow, I didn't know it. Thank you